Fix product categories list displaying [Object] instead of count

Aljullu opened the pull request:
I think I introduced this bug in #945. Product Categories List was displaying `[Object]` instead of the categories items count. This PR fixes that and gets rid of a `<span>`, which I think was not necessary.

### Screenshots
_Before:_
![image](https://user-images.githubusercontent.com/3616980/64439186-b4b0cb80-d0c9-11e9-9571-b7554e23b472.png)

_After:_
![image](https://user-images.githubusercontent.com/3616980/64439105-8df29500-d0c9-11e9-9c2e-673ada387709.png)

### How to test the changes in this Pull Request:

1. Create a post with a _Product Categories List_ block.
2. Make sure it displays the item count next to the category name.

### Changelog

> Make it easier for themes to style the Product Categories List block: new class names allow writing simpler selectors and it's now possible to remove the parentheses around the count number.

mikejolley commented:
@Aljullu The span is probably usefulf or styling? Without an element around the count you cannot for example align it away from the category name.

Aljullu commented:
> @Aljullu The span is probably useful for styling? Without an element around the count you cannot for example align it away from the category name.

@mikejolley good point! I added the `<span>` back. In order to make styling even easier, I also added class names to `<ul>` and `<li>` elements and moved the parentheses from the markup to the CSS, so themes can easily hide them.

Let me know how it looks.
